components/Heading/Heading.js

```diff
@@ -1,33 +1,67 @@
-import React from 'react';
-import { string, number, oneOfType, bool, node, oneOf } from 'prop-types';
+import React, { Component } from 'react';
+import { string, number, oneOfType, bool, oneOf } from 'prop-types';
 import classNames from 'classnames';
+import kebabCase from 'lodash/kebabCase';
+import ScreenReaderOnly from 'components/ScreenReaderOnly/ScreenReaderOnly';
+import LinkIcon from 'static/images/icons/FontAwesome/link-solid.svg';
 import styles from './Heading.css';
 
-Heading.propTypes = {
-  className: string,
-  id: oneOfType([string, number]), // reference for scroll anchors
-  children: node.isRequired,
-  hasHeadingLines: bool,
-  theme: oneOf(['gray', 'secondary', 'white']),
-};
+class Heading extends Component {
+  static propTypes = {
+    className: string,
+    id: oneOfType([string, number]), // reference for scroll anchors
+    customAnchorClass: string,
+    text: string.isRequired,
+    hasHeadingLines: bool,
+    hasHashLink: bool,
+    headingLevel: number,
+    theme: oneOf(['gray', 'secondary', 'white']),
+  };
 
-Heading.defaultProps = {
-  className: undefined,
-  id: '',
-  hasHeadingLines: true,
-  theme: 'gray',
-};
+  static defaultProps = {
+    className: undefined,
+    id: '',
+    customAnchorClass: 'anchorMargin',
+    hasHeadingLines: false,
+    hasHashLink: true,
+    headingLevel: 2,
+    theme: 'secondary',
+  };
 
-function Heading({ children, className, hasHeadingLines, id, theme }) {
-  const classes = classNames(className, styles.Heading, styles[theme], {
-    [`${styles.headingLines}`]: hasHeadingLines,
-  });
+  render() {
+    const { props } = this;
+    const anchorId = kebabCase(props.text);
+    const classes = classNames(styles[props.theme], {
+      [`${styles.Heading} ${props.className}`]: !props.hasHeadingLines,
+      [`${styles.headingOurMission}`]: props.className === 'headingOurMission',
+      [`${styles.headingLines}`]: props.hasHeadingLines,
+      [`${styles.headingTextWithLinkIconOffset}`]: props.hasHashLink,
+    });
+    const HeadingElement = `h${props.headingLevel}`;
 
-  return (
-    <h2 className={classes} id={id}>
-      {children}
-    </h2>
-  );
+    return (
+      <div className={`${styles.headingContainer}`}>
+        {props.hasHashLink && (
+          <span
+            id={anchorId}
+            className={classNames(styles.anchor, {
+              [styles.anchorMargin]: props.customAnchorClass === 'anchorMargin',
+              [styles.anchorMarginPodcast]: props.customAnchorClass === 'podcast',
+            })}
+          />
+        )}
+        {props.hasHashLink && (
+          <div className={`${styles.hashLinkContainer}`}>
+            <a href={`#${anchorId}`} data-testid="Hash Link">
+              <LinkIcon className={styles.icon} />
+              <ScreenReaderOnly>Scroll Link</ScreenReaderOnly>
+            </a>
+          </div>
+        )}
+        <HeadingElement className={classes}>{props.text}</HeadingElement>
```

Comment:
Can you nest the HeadingElement inside the anchor tag? It's good that we're letting DOM-focused individuals (keyboard users) know it's a scroll link, but we also need to give context of what it's scrolling to!

```
<div className={`${styles.hashLinkContainer}`}>
  <a href={`#${anchorId}`} data-testid="Hash Link">
    <LinkIcon className={styles.icon} />
    <ScreenReaderOnly>Scroll Link</ScreenReaderOnly>
    <HeadingElement className={classes}>{props.text}</HeadingElement>
  </a>
</div>
```

Reply:
I was having issues with wrapping this in the anchor tag. For example, my issue was that the link icon's right margin looked like it increased once the screen is at a small view port to where the heading wrapped into another line. Right now they are siblings and works. Hovering over the heading now displays the link icon.
![hoverheadingtextexample](https://user-images.githubusercontent.com/13701741/66276103-928fa200-e85d-11e9-82d9-ca69b2debd76.gif)

The current minor issue is the heading text wrapping again. However, this time its the additional line that is not centering. The heading current offsets the right margin so the text appears centered without the icon displaying. It looks fine with a single line. It has the centering issue with the wrapped line. I'm not sure how to fix this one.
![example2](https://user-images.githubusercontent.com/13701741/66276140-f5813900-e85d-11e9-8140-3643c5c188f7.png)
![multiline](https://user-images.githubusercontent.com/13701741/66276145-0467eb80-e85e-11e9-951f-1f0e0fb98c4d.png)